### DISCUSSION TITLE
Does Frigate cache wear out SSDs (Proxmox)?

### DISCUSSION THREAD

**Discussion by xekil** [blakeblackshear/frigate#13359] [General]
Hello,

I wonder if Frigate cache wears out SSDs?

Currently, Proxmox and my LXC Containers are on the same SSD (PNY 500GB), and the media folder is mounted on a USB HDD.

The SSD I use for Proxmox is only a year old and it already encounters I/O errors while its wear rate is only 2%. (I can no longer backup some LXC containers because of the errors).

I also have random complete freezes of the host with nothing in the logs (6 months, 1 month, 3 weeks).

I am thinking of replacing my SSD with a Samsung Pro model, but I wonder if it would not also be wise to move the location of the Frigate cache to the USB HDD?

- Is this possible?
- If yes, won't this slow down Frigate?

Or, although I don't like this idea (I fear reduced performance), would it be wiser to move the LXC Container to the HDD?

What is the best configuration to use with my hardware?

**Hardware:**

- HP EliteDesk 800 G3
- Proxmox 8.2.4
- i5 7600T 2.80GHz
- 32 GB RAM
- Container LXC Frigate 0.14.0, etc...
- SSD PNY 500Go M280CS2230-500-RB
- HDD USB WD 2To
- CORAL USB
- NAS storage (for Backup containers LXC and media for Jellyfin)

Thanks for your opinions

**Comment by NickM-27** [comment]
The Frigate cache (/tmp/cache) should always be using RAM not storage

I have frigate configured to write to an SSD cache pool (ZFS, Raid 1) for over a year now and it has been working flawlessly

**Reply by xekil** [reply]
Thanks for the reply, how to set this up?

**Reply by NickM-27** [reply]
the docs show how to configure it in docker, if you are using an LXC directly then you'll have to find that out elsewhere

**Reply by xekil** [reply]
For that?

>The Frigate cache (/tmp/cache) should always be using RAM not storage

**Reply by NickM-27** [reply]
yes https://docs.frigate.video/frigate/installation#storage

**Reply by xekil** [reply]
Thanks, I'm afraid it's not that easy to set up...
And regarding my initial question, should I deduce that yes "frigate wears out SSDs"?

**Reply by NickM-27** [reply]
writing to an SSD will "wear it out" just like writing anything else, there is nothing special about frigate that will make it wear more than something else writing as much data. You should use RAM for /tmp/cache so that only recordings that are actually going to be saved are written to the disk which will greatly reduce the amount of writes to the SSD

**Reply by xekil** [reply]
I think it might be wise to specify in the documentation with more insistence this point which seems important to me, in fact, on the internet or elsewhere, no one mentions this recommendation or seems to use it despite the multitude of tutorials that exist.

This will probably motivate competent people to detail the implementation procedure in the case of using Frigate LXC without docker.

So, I forget the migration of the LXC container to the HDD.

Thank you

**Reply by edalquist** [reply]
https://docs.frigate.video/frigate/installation/#storage tells you to use `tmpfs` and all the example docker configs on that page show using `tmpfs` as well.

Per the docker docs [tmpfs](https://docs.docker.com/engine/storage/tmpfs/) is an in-memory file system

**Reply by xekil** [reply]
Yes I saw that, thank you, but if you don't use docker there is nothing, I am now wondering what is the advantage of using LXC without docker :)

On the other hand, I don't understand, I have read several times on Reddit that by default Frigate uses RAM for the 10 second sequence without additional configuration, is this error of understanding of people who stipulate this? Thank you

**Reply by NickM-27** [reply]
it is enabled by default in the sense that HA addon is forced to work this way and the frigate documentation for installation (which is recommended via docker) has this in the default config. 

If something external and not officially supported like proxmox is used then yes, those users will need to figure out how their system requires them to configure it properly

**Reply by edalquist** [reply]
FYI Docker in LXC on Proxmox is totally doable as well, that is how I'm running Frigate. Would a tutorial post on how to set up the LXC container to successfully run docker + hardware acceleration + coral be helpful?

**Reply by NickM-27** [reply]
I believe there are already a few posts sharing how users accomplished that

**Reply by xekil** [reply]
> it is enabled by default in the sense that HA addon is forced to work this way and the frigate documentation for installation (which is recommended via docker) has this in the default config.
> 
> If something external and not officially supported like proxmox is used then yes, those users will need to figure out how their system requires them to configure it properly

Thanks for these details, I understand better now:

Add-on HA = Default RAM cache
Docker = Easily achievable RAM cache
LXC or Other = Achievable, but you have to find out how to do it

> FYI Docker in LXC on Proxmox is totally doable as well, that is how I'm running Frigate. Would a tutorial post on how to set up the LXC container to successfully run docker + hardware acceleration + coral be helpful?

Thanks, I have already used Frigate with Docker and indeed it is not the tutorial that is missing (I still have a container with Frigate 12 under docker) but I am looking to simplify my installations and limit the overlays, that is why I use Frigate 0.13.2 then 0.14.0 directly with LXC via the TTECK Script.
